Add guided Okta enrollment documentation

ptgott opened the pull request:
This change adds documentation for the guided Okta enrollment flow.

Closes #52821

kopiczko committed:
Docs: Okta integration docs overhaul

ptgott committed:
Edit the guided Okta enrollment documentation

- Flesh out the section introduction page in order to adopt a
  hub-and-spoke model for the section, in which each page is a how-to
  guide for setting up one component of the integration.

- Add the overview of integration components to the section introduction
  page. Also add information that pertains to the entire integration to
  the section overview, including information on the integration status
  page and deleting the integration.

- Organize appendix guides so the information within each one is part of
  a how-to guide for setting up an integration component.

- Add a separate guided SSO page so there is a how-to guide for each
  integration component.

- Minor readability edits, including removing some images.

- Remove unused images.

As with the original changes, sidebar items will be out of order. We can
edit the docs engine in a separate change to address this.

zmb3 commented on docs/img/enterprise/plugins/okta/okta-api-app-public-cluster-pub-key.png:
Should we crop the empty white space out of the right side of these images? It's nearly a third of the image.

zmb3 commented on docs/img/enterprise/plugins/okta/okta-delete-integration.png:
UX feedback: it's a little confusing to click a button that says Delete **Integration** to be presented with a confirmation dialog that asks if I want to  remove the _plugin_.

- Is there any particular reason to switch verbs from delete to remove?
- Is there any reason to switch nouns from integration to plugin?

ptgott replied:
Making sure @kopiczko sees this.

kopiczko replied:
Ping @kiosion

kiosion replied:
Not sure the reasoning; kept this as-is from the prior UI. Could update to be consistent, since in the UI we always refer to the Okta plugin as the 'Okta Integration'.

zmb3 commented on docs/pages/admin-guides/access-controls/sso/okta.mdx:
```diff
@@ -82,7 +82,70 @@ In this section we will create an application in the Okta dashboard to allow our
+
+#### General
+
+- Single sign on URL: `https://<cluster-url>:<port>/v1/webapi/saml/acs/okta`
```
Consider saying `cluster-hostname` or `cluster-endpoint` here.

The cluster "URL" would include https://..., and we don't want that twice.

zmb3 commented on docs/pages/admin-guides/access-controls/sso/okta.mdx:
```diff
@@ -82,7 +82,70 @@ In this section we will create an application in the Okta dashboard to allow our
+content from the group attribute statement.
+</Admonition>
+
+<Admonition type="tip">
```
Should we try to avoid stacking two admonitions on top of eachother?

zmb3 commented on docs/pages/enroll-resources/application-access/okta/app-and-group-sync.mdx:
```diff
@@ -0,0 +1,210 @@
+
+## How it works
+
+In Teleport, the Okta Access List synchronizer looks for any Okta group with
+members or any Okta application with individual assignments that matches filters
+that you can optionally supply during Okta integration enrollment. Access Lists
+will not be created for groups or applications without any assignments.
```
We described when access lists will _not_ be created, but we never said when they will be created. I'd try to rephrase this section.

zmb3 commented on docs/pages/enroll-resources/application-access/okta/user-sync.mdx:
Why does setting up Okta sync (a feature of Identity Governance) live under the "enrolling application resources" docs (a feature of Zero Trust Access)?

ptgott replied:
We've organized the Okta sync docs under "Application Access" since before this product division, and https://github.com/gravitational/teleport/pull/53510/commits/0ee67f8bc68fb530a5e0f2467da3c12f2105e6bc edited the docs without moving them. I'll find a new home for this (I think under "Access Controls", where the Access Requests/Lists docs live).

zmb3 reviewed (commented):
There are now a number of different ways to integrate Okta with Teleport. I think we need a high level overview page that explains the options and how to select which approach is best for your use case.

ptgott committed:
Respond to zmb3 feedback

- Crop white space from an image.
- Fix URL examples in the SSO setup instructions. Use a `Var` component
  to make this easier to modify.
- Remove unnecessary Admonition tags in the Okta SSO enrollment guide.
- Move the Okta sync section to Access Controls. This is a feature of
  Identity Governance, so it makes sense to place the docs alongside
  those of other Identity Governance features like Access Requests and
  Access Lists.
- Adjust redirects after moving the Okta section to Access Controls.
- Use a DocCardList for the Access Controls section. The new Okta
  section makes the index page quite long, so use a DocCardList instead
  of header sections.
- Lightly restructure the How it Works section of the App and Group Sync
  guide. Explain how the synchronizer creates Access Lists before
  indicating when it doesn't create them.

ptgott commented:
TODO: I still need to address this feedback:

> There are now a number of different ways to integrate Okta with Teleport. I think we need a high level overview page that explains the options and how to select which approach is best for your use case.

kiosion commented on docs/pages/admin-guides/access-controls/okta/app-and-group-sync.mdx:
```diff
@@ -0,0 +1,209 @@
+
+- An Okta organization with admin access.
+
+Enabling the Okta integration will make Teleport take ownership over app and
```
Now that Bidirectional Sync is implemented as well, might be worth adding wording for that here ("enabling the integration with bidirectional sync enabled...")

kiosion commented on docs/pages/admin-guides/access-controls/okta/app-and-group-sync.mdx:
```diff
@@ -0,0 +1,209 @@
+## Step 2/2. Configure user group and application imports
+
+After configuring default owners, you can customize which Okta user groups and
+applications should be imported. By default, all user groups and applications
```
The filters are a bit unintuitive UX-wise... If "sync all apps/groups" are unchecked but no filters are provided, this is treated the same as providing a wildcard `*`. Worth mentioning, imo

kiosion reviewed (approved):
LGTM, with a few comments

ptgott committed:
Respond to kiosion feedback

Include missing details.
